host/xtest/aes_perf.c

```diff
@@ -37,12 +37,116 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <tee_client_api.h>
+#include <tee_client_api_extensions.h>
 #include <time.h>
 #include <unistd.h>
 
 #include "ta_aes_perf.h"
 #include "crypto_common.h"
 
+#ifdef CFG_SECURE_DATA_PATH
+#include "ta_sdp_perf.h"
+#include "include/uapi/linux/ion.h"
+
+#define DEFAULT_ION_HEAP_TYPE	ION_HEAP_TYPE_UNMAPPED
+
+#if (TA_AES_ECB != TA_SDP_ECB || \
+	TA_AES_CBC != TA_SDP_CBC || \
+	TA_AES_CTR != TA_SDP_CTR || \
+	TA_AES_XTS != TA_SDP_XTS)
+#error "AES test TA and SDP test TA uses incompatible APIs"
+#endif
+
+/*
+ * Resources in case using secure buffers.
+ *
+ * SDP_BUFFER_DISABLE		buffer not allocated in secure share memory
+ * SDP_BUFFER_REGISTER		secure buffer, registered to TEE at TA invoc.
+ * SDP_BUFFER_PREREGISTERED	secure buffer, registered once to TEE.
```

Comment:
agree. I will replace `BUFFER_NON_SECURE` with `BUFFER_SHM_ALLOCATED` since the test explicitly allocate it.